File: 3-longest-substring-without-repeating-characters/3-longest-substring-without-repeating-characters.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        i = 0
        j = 0
        window = 0
        wordSet = dict()
        while j<len(s):
            if s[j] in wordSet:
                i = max(i, wordSet[s[j]])
            wordSet[s[j]]=j+1
            window=max(window, j-i+1)
            j+=1
        return window

File: 3-longest-substring-without-repeating-characters/3-longest-substring-without-repeating-characters.go
func lengthOfLongestSubstring(s string) int {
    wordMap:=make(map[byte]int)
    i:=0
    j:=0
    window:=0
    for j<len(s) {
        if val, ok:=wordMap[s[j]]; ok {
            i=max(i, val)
        }
        window=max(window, j-i+1)
        wordMap[s[j]]=j+1
        j++
    }
    return window
}

func max(i,j int) int {
    if i>j {
        return i
    }
    return j
}